### PR TITLE
Add plateau.v1 schedule

### PR DESCRIPTION
**Pull request by danieldk**
## Description

This schedule yields values from the wrapped schedule, exponentially scaled by the number of times optimization has plateaued.

### Types of change

Feature

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: website/docs/api-schedules.md [hunk 1]
```diff
@@ -375,3 +375,39 @@ period = 1000
 | `max_lr`   | <tt>float</tt> |
 | `period`   | <tt>int</tt>   |
 | **YIELDS** | <tt>float</tt> |
+
+## plateau {#cyclic_triangular tag="function" new="9"}
```
[the remark this answers is not in the document]
Comment (danieldk) [reply website/docs/api-schedules.md]: Fixed.

File: website/docs/api-schedules.md [hunk 2]
```diff
@@ -375,3 +375,39 @@ period = 1000
...
+@schedules = "plateau.v1"
+scale = 0.5
+max_patience = 2
+
+[learn_rate.shedule]
+@schedules = "constant.v1"
+rate = 1.0
+```
+
+</grid>
```
[the remark this answers is not in the document]
Comment (danieldk) [reply website/docs/api-schedules.md:414]: Fixed.

File: thinc/schedules.py [hunk 3]
```diff
@@ -155,6 +157,104 @@ def _clip(value: float, start: float, stop: float) -> float:
...
+
+    """Yields values from the wrapped schedule, exponentially scaled by the
+    number of times optimization has plateaued. This schedule requires an
+    additional last_score argument, which is a tuple of the shape
+    (last_score_step, last_score). This tuple indicates when a model was
+    last evaluated (last_score_step) and with what score (last_score).
+
+    max_patience (int): the number of evaluations without improvement when
+        we consider the model to have plateaued.
+    scale (float): scaling of the inner schedule (scale**n_plateaus * inner).
```
[the remark this answers is not in the document]
Comment (danieldk) [reply thinc/schedules.py:174]: Changed the other way around -- replace every reference to stagnations by plateaus.

File: thinc/schedules.py [hunk 4]
```diff
@@ -155,6 +157,104 @@ def _clip(value: float, start: float, stop: float) -> float:
...
+            "schedule": schedule,
+            "state": _PlateauState(
+                best_score=None, last_score_step=None, patience=0, stagnations=0
+            ),
+        },
+    )
+
+
+def _plateau_schedule(
+    schedule: Schedule, step: int, *, last_score: Tuple[int, float], **kwargs
```
[the remark this answers is not in the document]
Comment (danieldk) [reply thinc/schedules.py]: Fixed. I also changed the signature to use `None` as a default value. Then the schedule doesn't fail in weird ways if `last_score` is not provided.

File: thinc/schedules.py [hunk 5]
```diff
@@ -155,6 +157,104 @@ def _clip(value: float, start: float, stop: float) -> float:
     return max(value, stop) if (start > stop) else min(value, stop)
 
 
+@registry.schedules("plateau.v1")
+def plateau(
+    max_patience: int, scale: float, schedule: Schedule[float]
+) -> Schedule[float]:
+
+    """Yields values from the wrapped schedule, exponentially scaled by the
+    number of times optimization has plateaued. This schedule requires an
```
[the remark this answers is not in the document]
Comment (danieldk) [reply thinc/schedules.py]: Fixed.

**Review by shadeMe — APPROVED** [review]

**Pull request merged**